tests/report_test.py

```diff
@@ -0,0 +1,134 @@
+    report = create_report("shipment")
+
+    assert isinstance(report, easypost.Report)
+    assert str.startswith(report.id, "shprep_")
+
+
+@pytest.mark.vcr()
+def test_report_create_shipment_invoice_report():
+    report = create_report("shipment_invoice")
+
+    assert isinstance(report, easypost.Report)
+    assert str.startswith(report.id, "shpinvrep_")
+
+
+@pytest.mark.vcr()
+def test_report_create_tracker_report():
+    report = create_report("tracker")
+
+    assert isinstance(report, easypost.Report)
+    assert str.startswith(report.id, "trkrep_")
+
+
+@pytest.mark.vcr()
+def test_report_retrieve_payment_log_report():
+    report = create_report("payment_log")
+
+    retrieved_report = easypost.Report.retrieve(report.id)
+
+    assert isinstance(retrieved_report, easypost.Report)
+    assert report.start_date == retrieved_report.start_date
+    assert report.end_date == retrieved_report.end_date
+
+
+@pytest.mark.vcr()
+def test_report_retrieve_refund_report():
+    report = create_report("refund")
+
+    retrieved_report = easypost.Report.retrieve(report.id)
+
+    assert isinstance(retrieved_report, easypost.Report)
+    assert report.start_date == retrieved_report.start_date
+    assert report.end_date == retrieved_report.end_date
+
+
+@pytest.mark.vcr()
+def test_report_retrieve_shipment_report():
+    report = create_report("shipment")
+
+    retrieved_report = easypost.Report.retrieve(report.id)
+
+    assert isinstance(retrieved_report, easypost.Report)
+    assert report.start_date == retrieved_report.start_date
+    assert report.end_date == retrieved_report.end_date
+
+
+@pytest.mark.vcr()
+def test_report_retrieve_shipment_invoice_report():
+    report = create_report("shipment_invoice")
+
+    retrieved_report = easypost.Report.retrieve(report.id)
+
+    assert isinstance(retrieved_report, easypost.Report)
+    assert report.start_date == retrieved_report.start_date
+    assert report.end_date == retrieved_report.end_date
+
+
+@pytest.mark.vcr()
+def test_report_retrieve_tracker_report():
+    report = create_report("tracker")
+
+    retrieved_report = easypost.Report.retrieve(report.id)
+
+    assert isinstance(retrieved_report, easypost.Report)
+    assert report.start_date == retrieved_report.start_date
+    assert report.end_date == retrieved_report.end_date
+
+
+@pytest.mark.vcr()
+def test_report_all():
+    reports = easypost.Report.all(type="shipment", page_size=fixture.page_size())
+
+    reports_array = reports["reports"]
+
+    assert len(reports_array) <= fixture.page_size()
+    assert reports["has_more"] is not None
+    assert all(isinstance(report, easypost.Report) for report in reports_array)
+
+
+@pytest.mark.vcr()
+def test_report_create_no_type():
+    try:
+        _ = easypost.Report.create()
+    except Exception:
+        assert Exception
+
+
+@pytest.mark.vcr()
+def test_report_all_no_type():
+    try:
+        _ = easypost.Report.all()
+    except Exception:
+        assert Exception
```

Comment:
These tests aren't necessary. They were included in PHP for example because PHP had specific branching logic to that should have been tested whereas the Python lib does not have that.

Delete the cassettes along with this so they don't stay in the repo.

Reply:
removed